Tasks/KubernetesV1/src/kubernetes.ts

```diff
@@ -82,14 +99,72 @@ function executeKubectlCommand(clusterConnection: ClusterConnection, command: st
         "TaskEndpointId",
         "KubernetesV1",
         JSON.stringify(telemetry));
-    var result = "";
-    return commandImplementation.run(clusterConnection, command, (data) => result += data)
-    .fin(function cleanup() {
-        var commandOutputLength = result.length;
-        if (commandOutputLength > environmentVariableMaximumSize) {
-            tl.warning(tl.loc("OutputVariableDataSizeExceeded", commandOutputLength, environmentVariableMaximumSize));
-        } else {
-            tl.setVariable('KubectlOutput', result);
-        }
-    });
+
+    // The output result can contain more than one Json objects
+    // We want to parse each of the objects separately, hence push the output in JSON array form    
+    var result = [];
+    return commandImplementation.run(clusterConnection, command, (data) => result.push(data))
+        .fin(function cleanup() {
+            console.log("commandOutput" + result);
+            var commandOutputLength = result.length;
+            if (commandOutputLength > environmentVariableMaximumSize) {
+                tl.warning(tl.loc("OutputVariableDataSizeExceeded", commandOutputLength, environmentVariableMaximumSize));
+            } else {
+                tl.setVariable('KubectlOutput', result.toString());
+            }
+
+            if (addPipelineMetadata && addPipelineMetadata.toLowerCase() == "true") {
+                // For each output, check if it contains a JSON object
+                result.forEach(res => {
```

Comment:
outputFormat may not always Json. If it is not json we are not capturing the data. is there anyway to capture the data if outputformat not specified or it is yaml.

Reply:
For deploy commands like apply, the output format is always either JSON or YAML, added checks for parsing data for both these formats.